### DISCUSSION TITLE
Disable shorthands

### DISCUSSION THREAD

**Discussion by apih** [alpinejs/alpine#4125] [1. Help]
Hi. Is it possible to disable shorthand syntax for `x-bind:` (`:`) and `x-on:` (`@`)?

**Comment by SimoTod** [comment]
No, it's not possible.

**Comment by ekwoka** [comment]
Is there a specific reason you'd want to?

**Reply by apih** [reply]
I use Alpine with Laravel Blade, so just want to enforce a standardization.

**Reply by ekwoka** [reply]
Seems like the time to just use Livewire 😂

**Reply by joshhanley** [reply]
This might be why 😂 

<img width="627" alt="image" src="https://github.com/alpinejs/alpine/assets/882837/960e3421-0d7d-4023-b411-fe2306434f3c">

I wouldn't mind it myself. Livewire views are just blade, so using Blade and Alpine together with Alpine shorthands isn't a great experience. Would be nice to be able to enforce in a project that shorthand isn't allowed. Haven't looked into it yet though.

**Reply by ekwoka** [reply]
I'm curious what the actual issue is when using blade but not livewire that makes the problem...

**Reply by joshhanley** [reply]
@ekwoka blade and Livewire both have the problem as Livewire's templating engine is Blade. Technically speaking there is no error, it just increases the likelyhood of human error.

In Blade (and as such Livewire), you use `@` everywhere to denote blade directives and `:` is used to pass executable code into a blade component.

For example, you could have a button with a conditionally rendered `disabled` in Blade like this:

```blade
<button @if(true) disabled @endif>
```

But if you also have click listeners on it, like

```blade
<button @if(true) disabled @endif @click="doSomething">
```

It just doesn't read great (especially for beginners) as it's hard to determine what is Alpine and what is Blade.

And in the case of a blade component, a lot of time with Livewire you will have a blade component that wraps up an Alpine component. 

So you might use it like this:

```blade
<x-my-blade-component :class="{ 'myclass': true}">
```

The problem with the above is Blade tries to execute everything inside of the class attribute as PHP, because `:` denotes that the code passed in is php (like a variable or something). So in the example below, `$myVariable` will be passed through to the component as a prop rather than being converted to a string due to the `:`

```blade
<x-my-blade-component :someProp="$myVariable">
```

To allow Alpine short syntax to work with Blade components, Laravel does have the `::` option, which sort of escapes the `:` shorthand for `x-bind`. So to get that first class example to work it would look like this:

```blade
<x-my-blade-component ::class="{ 'myclass': true}">
```

Again not huge problems but my personal preference would be to set a standard in my projects that only `x-on:` and `x-bind:` work and not the shorthands.

**Reply by ekwoka** [reply]
Ah, yeah, that makes a lot of sense.

Interesting that Caleb decided to provide them at all under this case, even if it was to imitate the Vue API....

**Reply by apih** [reply]
Good explanation by @joshhanley. When onboarding junior developers into the team, the usage of shorthands do cause confusions and humar errors. If @calebporzio can add a way to disable the usage of shorthands, it would be a good.

**Reply by SimoTod** [reply]
What benefit would you have with disabling it? 

I'm not sure how disabling the option in Alpine would stop accidental human mistakes and blade picking up the directive since Blade runs first.

If your team knows not to use it, it's not a problem, they just won't use it. If you onboard someone new and they add an alpine directive using the shortand version in a blade template because they just cclopy pasted from somewhere, blade can still interpretate it as its own directive by mistake even if it's later on disabled in Alpine.
There won't be anything flagging it as an obvious error. Probably best if you come up with some sort of linting rules at that point.
